torch/_inductor/codegen/wrapper.py

```diff
@@ -744,16 +744,28 @@ def memory_plan_reuse(self):
             self.lines.pop()
 
         # codegen allocations in two passes
-        planning_state = MemoryPlanningState()
+        planning_states = [MemoryPlanningState()]
+        past_planning_states = []
         for i in range(len(self.lines)):
             line = self.lines[i]
             if isinstance(line, MemoryPlanningLine):
-                self.lines[i] = line.plan(planning_state)
+                self.lines[i] = line.plan(planning_states[-1])
+            elif isinstance(line, EnterScopeLine):
+                planning_states.append(MemoryPlanningState())
+            elif isinstance(line, ExitScopeLine):
+                past_planning_states.append(planning_states.pop())
+        past_planning_states.append(planning_states.pop())
```

Comment:
nit - could we add an assert to ensure we have a single ```MemoryPlanningState``` instance after the loop? Just to make sure we have paired enter/exit states. Thanks.